### PR TITLE
docs: Added geomap missing headings, why and how describing data needs

### PR DESCRIPTION
**Pull request by srperf**
**What is this feature?**

This PR adds some formatting and complements descriptions on why the user could wish to use the visualization, as well as explanations on how focusing on data needs and examples.

**Why do we need this feature?**

Users will get better insights into why they might need this visualization and how to get on working with it.

**Who is this feature for?**

Every user looking for instructions and info to start working with geo maps.

**Which issue(s) does this PR fix?**:

None

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 1]
```diff
@@ -49,20 +49,95 @@ refs:
 
 # Geomap
 
-Geomaps allow you to view and customize the world map using geospatial data. You can configure various overlay styles and map view settings to easily focus on the important location-based characteristics of the data.
+Geomaps allow you to view and customize the world map using geospatial data. It's the ideal visualization if you have data that includes location information and you want to see it displayed in a map. You can configure various overlay styles, like routes, as well as map view settings including adding numeric values, labels, and icons. This helps you to easily focus on the important location-based characteristics of the data.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:52]:
```suggestion
Geomaps allow you to view and customize the world map using geospatial data. It's the ideal visualization if you have data that includes location information and you want to see it displayed in a map. You can configure and overlay various types of layers, like markers, heatmaps, routes, and network. In this mix, you can also overlay and blend various basemaps we include or your own custom map. You can customize the map view extents and even set it to fit your data dynamically. This helps you to easily focus on the important location-based characteristics of the data.
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 2]
```diff
@@ -49,20 +49,95 @@ refs:
...
 
-> We would love your feedback on geomaps. Please check out the [open Github issues](https://github.com/grafana/grafana/issues?page=1&q=is%3Aopen+is%3Aissue+label%3Aarea%2Fpanel%2Fgeomap) and [submit a new feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=type%2Ffeature-request,area%2Fpanel%2Fgeomap&title=Geomap:&projects=grafana-dataviz&template=1-feature_requests.md) as needed.
+{{< figure src="/static/img/docs/geomap-panel/geomap-example-8-1-0.png" max-width="1200px" alt="Geomap visualization" >}}
 
-{{< figure src="/static/img/docs/geomap-panel/geomap-example-8-1-0.png" max-width="1200px" caption="Geomap panel" >}}
+When a geomap is in focus, you can pan around using the arrow keys or zoom in and out using the plus (`+`) and minus (`-`) keys or icons.
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:56]:
```suggestion
When a geomap is in focus, in addition to typical mouse controls, you can pan around using the arrow keys or zoom in and out using the plus (`+`) and minus (`-`) keys or icons.
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 3]
```diff
@@ -49,20 +49,95 @@ refs:
...
 
-Pan the map, while it's in focus, by using the arrow keys. Zoom in and out by using the `+` and `-` keys.
+Geomaps are also useful when you have location data that’s changing in real time and you want to visualize where an element is moving, using auto-refresh.
+
+You can use a geomap visualization if you need to:
+
+- Track the position of food deliveries, planes, or freight services
+- Show the locations of the data centers or corporate devices
+- Map hiking route
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:64]:
```suggestion
- Track your fleet of vehicles and associated metrics
- Show the locations and statuses of data centers or other connected assets in a network
- Display geographic trends in a heatmap
- Visualize the relationship of your locations' HVAC consumption or solar production with the sun's location
```

File: docs/sources/panels-visualizations/visualizations/geomap/index.md [hunk 4]
```diff
@@ -49,20 +49,95 @@ refs:
...
+| 2016 | MEX    | 104171935 |
+| 2016 | DEU    | 94393454  |
+| 2016 | FRA    | 83654250  |
+| 2016 | BRA    | 80921527  |
+| 2016 | CAN    | 79699762  |
+
+The other location types&mdash; airport codes or US state codes&mdash;aren't automatically detected.
+
+If you want to use other codes or give the field a custom name, you can follow the steps in the [Location mode](#location-mode) section.
+
```
Comment (drew08t) [inline docs/sources/panels-visualizations/visualizations/geomap/index.md:140]: These are great examples! Is the intention to have these at the top and have the users dig deeper down below if needed?

Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/geomap/index.md:140]: @srperf FTW! Definitely intended to be down here.

File: docs/sources/panels-visualizations/visualizations/geomap/index.md (hunk 1 above)
Comment (imatwawana) [reply docs/sources/panels-visualizations/visualizations/geomap/index.md:52]: I'm wary of overloading this opening section. When you look at it in preview, it becomes a fair bit to read before the next heading. What about:
```suggestion
Geomaps allow you to view and customize the world map using geospatial data. It's the ideal visualization if you have data that includes location information and you want to see it displayed in a map.

You can configure and overlay [map layers](#types), like markers and routes, and blend included basemaps or your own custom maps. This helps you to easily focus on the important location-based characteristics of the data.
```

Comment (drew08t) [reply docs/sources/panels-visualizations/visualizations/geomap/index.md:52]: That is much better. One suggestion I have is to mention "heatmaps and networks" as these are more interesting or broadly applicable use cases imo.

**Review by imatwawana — APPROVED** [review]

**Pull request merged**